client/app/bundles/course/assessment/submission/pages/SubmissionEditIndex/SubmissionEditForm.jsx

```diff
@@ -538,44 +545,51 @@ const SubmissionEditForm = (props) => {
     />
   );
 
-  const renderSteppedQuestions = () => (
-    <Stepper
-      activeStep={stepIndex}
-      connector={<div />}
-      nonLinear
-      style={{ justifyContent: 'center', flexWrap: 'wrap', padding: 10 }}
-    >
-      {questionIds.map((id, index) => {
-        let stepButtonColor = '';
-        const isCurrentQuestion = index === stepIndex;
-        stepButtonColor = isCurrentQuestion ? blue[800] : blue[400];
-        return (
-          <Step key={id} sx={{ width: 55, height: 50 }}>
-            <StepButton
-              icon={
-                <SvgIcon htmlColor={stepButtonColor}>
-                  <circle cx="12" cy="12" r="12" />
-                  <text
-                    fill="#fff"
-                    fontSize="12"
-                    textAnchor="middle"
-                    x="12"
-                    y="16"
-                  >
-                    {index + 1}
-                  </text>
-                </SvgIcon>
-              }
-              onClick={() => {
-                setStepIndex(index);
-              }}
-              style={styles.stepButton}
-            />
-          </Step>
-        );
-      })}
-    </Stepper>
-  );
+  const renderSteppedQuestions = () => {
+    const answerId = getCurrentlySavingAnswerId();
```

Comment:
This `answerId = getCurrentlySavingAnswerId()` shouldn't be outside the loop outside of Stepper.
This setup forces any change in answerId will trigger a re-render of ALL the stepper bubbles.

Inside the `questionIds.map((id, index) => ...` we already have the associated `questionId`, so inside the `onClick` event we can obtain `questions[id].answerId` directly.

Reply:
understood. I moved out the saving instruction into the function instead, so that the fetching of the answerId will be done only when user click on the bubble (so it won't get re-rendered)